### PR TITLE
feat: Add functions for PanelEvent.OPEN

### PR DESCRIPTION
**Pull request by mofojed**
- Instead of using PanelEvent.OPEN directly and not getting any type safety, add functions for emitting, listening, and a hook to enforce type safety
- Add xcomponents to the styleguide

### CODE REVIEW

File: packages/golden-layout/src/utils/EventUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    jest.clearAllMocks();
+  });
+
+  it('listenForEvent', () => {
+    const event = 'test';
+    const handler = jest.fn();
+    const remove = listenForEvent(eventEmitter, event, handler);
+    expect(eventEmitter.on).toHaveBeenCalledWith(event, handler);
+    remove();
+    expect(eventEmitter.off).toHaveBeenCalledWith(event, handler);
```
Comment (mattrunyon) [inline packages/golden-layout/src/utils/EventUtils.test.ts:25]: Check off isn't called before the `remove` call just to verify

File: packages/golden-layout/src/utils/EventUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    expect(eventEmitter.on).toHaveBeenCalledWith(event, handler);
+  });
+
+  it('makeEmitFunction', () => {
+    const event = 'test';
+    const emit = makeEmitFunction(event);
+    const payload = { test: 'test' };
+    emit(eventEmitter, payload);
+    expect(eventEmitter.emit).toHaveBeenCalledWith(event, payload);
+  });
```
Comment (mattrunyon) [inline packages/golden-layout/src/utils/EventUtils.test.ts:42]: Add a test for the hook? And that it removes/adds new when the event hub or callback changes

File: packages/golden-layout/src/utils/index.ts [hunk 3]
```diff
@@ -6,3 +6,4 @@ export { default as ReactComponentHandler } from './ReactComponentHandler';
 export * from './ConfigMinifier';
 export { default as BubblingEvent } from './BubblingEvent';
 export { default as EventHub } from './EventHub';
+export * from './EventUtils';
```
Comment (mattrunyon) [inline packages/golden-layout/src/utils/index.ts:9]: Do we want to export everything here or just `makeEventFunctions` and the types?

**Review by mattrunyon — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/PanelEvent.ts [hunk 4]
```diff
@@ -7,16 +7,26 @@ export type WidgetDescriptor = {
 };
 
 export type PanelOpenEventDetail<T = unknown> = {
-  dragEvent?: DragEvent;
-  fetch?: () => Promise<T>;
+  /** Opening the widget was triggered by dragging from a list. The coordinates are used as the starting location for the drag.  */
+  dragEvent?: MouseEvent;
```
[the remark this answers is not in the document]
Comment (mofojed) [reply packages/dashboard/src/PanelEvent.ts:11]: When you're dragging from the widget list to add a widget to the dashboard, that's what this drag event is for.

File: packages/golden-layout/src/utils/index.ts (hunk 3 above)
Comment (mofojed) [reply packages/golden-layout/src/utils/index.ts:9]: I don't see any reason _not_ to export the other functions.

Comment (mattrunyon) [reply packages/golden-layout/src/utils/index.ts:9]: My thought was if we expect to use `makeEventFunctions` and not the specific function calls then might as well only export that, but it doesn't hurt anything to export the others

**Pull request merged**